dm/unit.go

```diff
@@ -116,24 +151,44 @@ func (u *unitHolder) tryUpdateStatus(ctx context.Context, base lib.BaseWorker) e
 
 	switch strategy {
 	case worker.ResumeSkip:
+		// update status on first skip
+		if u.lastStage == worker.ResumeSkip {
+			return nil
+		}
+		u.unit.Pause()
+		u.lastStage = worker.ResumeSkip
 		// wait on next auto resume
+		status.Stage = metadata.StagePaused
+		statusBytes, err := json.Marshal(status)
+		if err != nil {
+			return err
+		}
 		s := libModel.WorkerStatus{
 			Code:         libModel.WorkerStatusError,
 			ErrorMessage: unit.JoinProcessErrors(result.Errors),
+			ExtBytes:     statusBytes,
 		}
 		// TODO: UpdateStatus too frequently?
 		// nolint:errcheck
 		_ = base.UpdateStatus(ctx, s)
 		return nil
 	case worker.ResumeDispatch:
+		u.unit.Pause()
```

Comment:
maybe we can check if the unit has been paused by ResumeSkip. But it should also be OK that we Pause twice